hasjob/templates/domain.html

```diff
@@ -12,20 +12,63 @@
                 <img itemprop="image" alt="Logo" src="{{ domain.logo_url }}"/>
               </div>
               <div class="col-xs-12 col-sm-9 col-sm-pull-3">
+          {%- elif domain.post_with_logo() %}
+            <div class="row">
+              <div class="img-container col-xs-12 col-sm-3 col-sm-push-9 post-company-logo">
+                <img itemprop="image" alt="Logo" src="{{ domain.post_with_logo().url_for('logo', _external=true) }}"/>
```

Comment:
Move this logic out into the Domain model. Add a method that returns a logo (from the domain itself or from the post). Decorate with `@cached_property` so you don't hit the database twice.